### DISCUSSION TITLE
best dev practices without autoreload?

### DISCUSSION THREAD

**Discussion by ncullen93** [ThinkR-open/golem#897] [Q&A]
Hi all, I'm wondering if anyone can share tips on how to best develop Shiny app using the golem framework while not being able to use autoreload. It seems like autoreload is so vital to Shiny development and not having it has made it hard for me to adopt golem - especially early on in projects when I am experimenting. 

Do people just have insane discipline and stop + restart their app every time they make a small change, or do people opt for test-driven development? What about UI decisions - what is the best way to test small changes without accumulating 100 dead tabs in the browser?

Interested to hear any insights or a pointer to where I can read more about this. Thanks for the feedback!

**Comment by ColinFay** [comment]
Hey, 

That's been a long-wanted feature of `{golem}` indeed. 

Several efficient approaches are just a keystroke away:

- There is an RStudio addin that comes with `{golem}` that launched `golem::run_dev()`. You can map it to a specific shortcut, then it's "Esc" + "Your keyboard shortcut" to relaunch the app 
- If you're a VSCode user like me, here is my custom keyboard shortcut: 

```
    {
        "description": "Golem run dev",
        "key": "Ctrl+Cmd+C",
        "command": "r.runCommand",
        "args": "golem::run_dev()"
    }
```

=> I've chosen this shortcut as it allows me to kill the running app with ctrl + C then run dev with ctrl + cmd + C, so it's pretty fast. 

So, even without the autoreload, and given that run_dev() will save all the open docs, and do an efficient clean up of your search path (which shiny options doesn't do), you can get pretty efficient, and reload the app quickly.
